diff --git a/core/psa_crypto_driver_wrappers.h b/core/psa_crypto_driver_wrappers.h
index 9891d2ced..3643ec1f0 100644
--- a/core/psa_crypto_driver_wrappers.h
+++ b/core/psa_crypto_driver_wrappers.h
@@ -45,6 +45,11 @@
 #if defined(MBEDTLS_PSA_P256M_DRIVER_ENABLED)
 #include "../drivers/p256-m/p256-m_driver_entrypoints.h"
 
+#endif
+/* Headers for the ESP32 ECC accelerator transparent driver */
+#if defined(MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED)
+#include "esp_ecc_psa_driver.h"
+
 #endif
 
 /* END-driver headers */
@@ -286,6 +291,26 @@ static inline psa_status_t psa_driver_wrapper_sign_hash(
                 return( status );
             }
 #endif /* MBEDTLS_PSA_P256M_DRIVER_ENABLED */
+#if defined (MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED)
+            if( PSA_KEY_TYPE_IS_ECC( psa_get_key_type(attributes) ) &&
+                PSA_ALG_IS_RANDOMIZED_ECDSA(alg) &&
+                PSA_KEY_TYPE_ECC_GET_FAMILY(psa_get_key_type(attributes)) == PSA_ECC_FAMILY_SECP_R1 &&
+                psa_get_key_bits(attributes) == 256 )
+            {
+                status = esp_ecc_transparent_sign_hash( attributes,
+                                                        key_buffer,
+                                                        key_buffer_size,
+                                                        alg,
+                                                        hash,
+                                                        hash_length,
+                                                        signature,
+                                                        signature_size,
+                                                        signature_length );
+                /* Declared with fallback == true */
+                if( status != PSA_ERROR_NOT_SUPPORTED )
+                    return( status );
+            }
+#endif /* MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED */
 #endif /* PSA_CRYPTO_ACCELERATOR_DRIVER_PRESENT */
             /* Fell through, meaning no accelerator supports this operation */
             return( psa_sign_hash_builtin( attributes,
@@ -368,6 +393,25 @@ static inline psa_status_t psa_driver_wrapper_verify_hash(
                 return( status );
             }
 #endif /* MBEDTLS_PSA_P256M_DRIVER_ENABLED */
+#if defined (MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED)
+            if( PSA_KEY_TYPE_IS_ECC( psa_get_key_type(attributes) ) &&
+                PSA_ALG_IS_ECDSA(alg) &&
+                PSA_KEY_TYPE_ECC_GET_FAMILY(psa_get_key_type(attributes)) == PSA_ECC_FAMILY_SECP_R1 &&
+                psa_get_key_bits(attributes) == 256 )
+            {
+                status = esp_ecc_transparent_verify_hash( attributes,
+                                                          key_buffer,
+                                                          key_buffer_size,
+                                                          alg,
+                                                          hash,
+                                                          hash_length,
+                                                          signature,
+                                                          signature_length );
+                /* Declared with fallback == true */
+                if( status != PSA_ERROR_NOT_SUPPORTED )
+                    return( status );
+            }
+#endif /* MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED */
 #endif /* PSA_CRYPTO_ACCELERATOR_DRIVER_PRESENT */
 
             return( psa_verify_hash_builtin( attributes,
@@ -726,6 +770,20 @@ static inline psa_status_t psa_driver_wrapper_generate_key(
                 }
 
 #endif /* MBEDTLS_PSA_P256M_DRIVER_ENABLED */
+#if defined(MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED)
+                if( PSA_KEY_TYPE_IS_ECC( psa_get_key_type(attributes) ) &&
+                    psa_get_key_type(attributes) == PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1) &&
+                    psa_get_key_bits(attributes) == 256 )
+                {
+                    status = esp_ecc_transparent_generate_key( attributes,
+                                                               key_buffer,
+                                                               key_buffer_size,
+                                                               key_buffer_length );
+                    if( status != PSA_ERROR_NOT_SUPPORTED )
+                        break;
+                }
+
+#endif /* MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED */
             }
 #endif /* PSA_CRYPTO_ACCELERATOR_DRIVER_PRESENT */
 
@@ -2725,6 +2783,25 @@ static inline psa_status_t psa_driver_wrapper_key_agreement(
                     return( status );
             }
 #endif /* MBEDTLS_PSA_P256M_DRIVER_ENABLED */
+#if defined(MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED)
+            if( PSA_KEY_TYPE_IS_ECC( psa_get_key_type(attributes) ) &&
+                PSA_ALG_IS_ECDH(alg) &&
+                PSA_KEY_TYPE_ECC_GET_FAMILY(psa_get_key_type(attributes)) == PSA_ECC_FAMILY_SECP_R1 &&
+                psa_get_key_bits(attributes) == 256 )
+            {
+                status = esp_ecc_transparent_key_agreement( attributes,
+                                                            key_buffer,
+                                                            key_buffer_size,
+                                                            alg,
+                                                            peer_key,
+                                                            peer_key_length,
+                                                            shared_secret,
+                                                            shared_secret_size,
+                                                            shared_secret_length );
+                if( status != PSA_ERROR_NOT_SUPPORTED)
+                    return( status );
+            }
+#endif /* MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED */
 #endif /* PSA_CRYPTO_ACCELERATOR_DRIVER_PRESENT */
 
             /* Software Fallback */
diff --git a/core/psa_crypto_driver_wrappers_no_static.c b/core/psa_crypto_driver_wrappers_no_static.c
index a8bd7eb3a..ee2e0c36d 100644
--- a/core/psa_crypto_driver_wrappers_no_static.c
+++ b/core/psa_crypto_driver_wrappers_no_static.c
@@ -43,6 +43,11 @@
 #if defined(MBEDTLS_PSA_P256M_DRIVER_ENABLED)
 #include "../drivers/p256-m/p256-m_driver_entrypoints.h"
 
+#endif
+/* Headers for the ESP32 ECC accelerator transparent driver */
+#if defined(MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED)
+#include "esp_ecc_psa_driver.h"
+
 #endif
 
 /* END-driver headers */
@@ -159,6 +164,20 @@ psa_status_t psa_driver_wrapper_export_public_key(
                 return( status );
 #endif
 
+#if (defined(MBEDTLS_PSA_ESP_ECC_DRIVER_ENABLED) )
+            status = esp_ecc_transparent_export_public_key
+                (attributes,
+                                key_buffer,
+                                key_buffer_size,
+                                data,
+                                data_size,
+                                data_length
+            );
+
+            if( status != PSA_ERROR_NOT_SUPPORTED )
+                return( status );
+#endif
+
 
 #endif /* PSA_CRYPTO_ACCELERATOR_DRIVER_PRESENT */
             /* Fell through, meaning no accelerator supports this operation */
